File: plant-back/PlantLibrary/Models/Plant.cs
using SharedLibrary.Database;
using System.Collections.Generic;

namespace PlantLibrary.Models
{
    [BsonCollection("Plant")]
    public class Plant : Document
    {
        public string DefaultName { get; set; }

        public string Family { get; set; }
        public string Genus { get; set; }
        public string Species { get; set; }
        public string Cultivar { get; set; }
        public string Variety { get; set; }
        public string ScientificName { get; set; }
        public List<string> CommonNames { get; set; }
        public string Description { get; set; }
        public string LocationOrigin { get; set; }

        public string TemperatureCMin { get; set; }
        public string TemperatureCMax { get; set; }
        public string TemperatureCIdeal { get; set; }
        public string Brightness { get; set; }
        public string Humidity { get; set; }
        public string Substrate { get; set; }
        public string SoilPH { get; set; }
        public string Multiplication { get; set; }
        public string Wintering { get; set; }

        public List<Picture> Pictures { get; set; }

    }
}

File: plant-front/src/models/plant.ts
export interface IPlant {
  id: number;
  family: string;
  genus: string;
  species: string;
  cultivar: string;
  variety: string;
  scientificName: string;
  commonNames: string[];
  description: string;
  locationOrigin: string;
  temperatureCMin: string;
  temperatureCMax: string;
  temperatureCIdeal: string;
  brightness: string;
  humidity: string;
  substrate: string;
  soilPH: string;
}

export class Plant {
  Id: number;
  Brightness: string;
  CommonNames: string[];
  Cultivar: string;
  Description: string;
  Family: string;
  Genus: string;
  Humidity: string;
  LocationOrigin: string;
  ScientificName: string;
  SoilPH: string;
  Species: string;
  Substrate: string;
  TemperatureCMax: string;
  TemperatureCMin: string;
  TemperatureCIdeal: string;
  Variety: string;

  constructor(input: IPlant) {
    this.Id = input.id;
    this.Brightness = input.brightness;
    this.CommonNames = input.commonNames;
    this.Cultivar = input.cultivar;
    this.Description = input.description;
    this.Family = input.family;
    this.Genus = input.genus;
    this.Humidity = input.humidity;
    this.LocationOrigin = input.locationOrigin;
    this.ScientificName = input.scientificName;
    this.SoilPH = input.soilPH;
    this.Species = input.species;
    this.Substrate = input.substrate;
    this.TemperatureCMax = input.temperatureCMax;
    this.TemperatureCMin = input.temperatureCMin;
    this.TemperatureCIdeal = input.temperatureCIdeal;
    this.Variety = input.variety;
  }

  getSearchValue(): string {
    let str = this.ScientificName;

    for (const cn of this.CommonNames) {
      str += ' ' + cn;
    }

    return str;
  }

}
